activation/activation.go

```diff
@@ -897,20 +911,23 @@ func (b *Builder) getPositioningAtx(
 	publish types.EpochID,
 	previous *types.ActivationTx,
 ) (types.ATXID, error) {
-	id, err := b.searchPositioningAtx(ctx, nodeID, publish)
+	ctxWithTimeout := ctx
+	var cancel context.CancelFunc
+
+	if b.poetCfg.PositioningATXSelectionTimeout > 0 {
+		ctxWithTimeout, cancel = context.WithTimeout(ctx, b.poetCfg.PositioningATXSelectionTimeout)
+		defer cancel()
+	}
+
+	id, err := b.searchPositioningAtx(ctxWithTimeout, nodeID, publish, previous)
 	if err != nil {
 		return types.EmptyATXID, err
 	}
 
-	if previous != nil {
-		switch {
-		case id == b.conf.GoldenATXID:
-			id = previous.ID()
-		case id != b.conf.GoldenATXID:
-			if candidate, err := atxs.Get(b.db, id); err == nil {
-				if previous.TickHeight() >= candidate.TickHeight() {
-					id = previous.ID()
-				}
+	if previous != nil && !bytes.Equal(id.Bytes(), previous.ID().Bytes()) {
```

Comment:
This creates copies of both `id` and `previous` just to compare them. How about

```suggestion
	if previous != nil && id != *previous {
```